srml/contract/src/rent.rs

```diff
@@ -19,80 +19,99 @@ use runtime_primitives::traits::{As, Bounded, CheckedDiv, CheckedMul, Saturating
 use srml_support::traits::{Currency, ExistenceRequirement, Imbalance, WithdrawReason};
 use srml_support::StorageMap;
 
+#[derive(PartialEq, Eq, Copy, Clone)]
+#[must_use]
+pub enum RentOutcome {
+	/// Exempted from rent iff:
+	/// * rent is offset completely by the `rent_deposit_offset`,
+	/// * or rent has already been paid for this block number,
+	/// * or account doesn't have a contract,
+	/// * or account has a tombstone.
+	Exempted,
+	/// Evicted iff:
+	/// * rent exceed rent allowance,
+	/// * or can't withdraw the rent,
+	/// * or go below subsistence threshold.
+	Evicted,
+	/// The outstanding dues were paid.
+	Paid,
+}
+
 /// Evict and optionally pay dues (or check account can pay them otherwise), for given block number.
```

Comment:
hmm either we say given block number and pass the block number. Butas we pass handicap now we should rephrase: at the block number current - handicap or something. All in all I'm not sure it is less confusing with handicap :-/

Reply:
then the handicap must be documented

Reply:
> then the handicap must be documented

Yep, just pushed that.